Mage.Tests/src/test/java/org/mage/test/cards/single/cmm/DemonOfFatesDesignTest.java

```diff
@@ -0,0 +1,262 @@
+package org.mage.test.cards.single.cmm;
+
+import mage.constants.PhaseStep;
+import mage.constants.Zone;
+import org.junit.Test;
+import org.mage.test.serverside.base.CardTestPlayerBase;
+
+/**
+ * @author Susucr
+ */
+public class DemonOfFatesDesignTest extends CardTestPlayerBase {
+
+    // Demon of Fate's Design
+    // {4}{B}{B}
+    // Enchantment Creature — Demon
+    //
+    // Flying, trample
+    // Once during each of your turns, you may cast an enchantment spell by paying life equal to its mana value rather than paying its mana cost.
+    // {2}{B}, Sacrifice another enchantment: Demon of Fate’s Design gets +X/+0 until end of turn, where X is the sacrificed enchantment’s mana value.
+    private static final String demon = "Demon of Fate's Design";
+
+    @Test
+    public void CastForLife() {
+        setStrictChooseMode(true);
+
+        addCard(Zone.BATTLEFIELD, playerA, demon);
+
+        addCard(Zone.HAND, playerA, "Glorious Anthem"); // Enchantment {1}{W}{W}
+
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Glorious Anthem");
+        setChoice(playerA, true); // yes to alt cast
+
+        setStopAt(1, PhaseStep.BEGIN_COMBAT);
+        execute();
+
+        assertPermanentCount(playerA, "Glorious Anthem", 1);
+        assertLife(playerA, 20 - 3);
+    }
+
+    @Test
+    public void SayNoToTheDemon() {
+        setStrictChooseMode(true);
+
+        addCard(Zone.BATTLEFIELD, playerA, demon);
+
+        addCard(Zone.HAND, playerA, "Glorious Anthem"); // Enchantment {1}{W}{W}
+
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Glorious Anthem");
+        setChoice(playerA, false); // no to alt cast
+
+        boolean hadError = false;
+        try {
```

Comment:
does `checkPlayableAbility()` not work here?

Reply:
So what is happening here is:
-> You can use the Demon's ability to cast Glorious Anthem.
-> You do not have mana to cast Anthem the normal way.

On casting the Anthem, there is a choice for 'do you want to use the alternate way from the Demon' with a yes/no answer.
-> On 'Yes', the spell is cast, life is lost as a cost.
-> On 'No', the engine attempts to use the normal cast command, but that is not available.

I am not sure how to clean the alternate cast to cancel the original cast on answering 'No' (or even if that should be handled that way). That test is just documenting that this is the current behavior.
With manual testing, the cast is cancelled, the card goes back in hand, and the Demon's ability is still avaible.

Reply:
I see, thanks for explaining